### DISCUSSION TITLE
Use basPath, rewrites, and getServerSideProps together cause page to reload

### DISCUSSION THREAD

**Discussion by DumbledoreD** [vercel/next.js#14536] [Help] [answered]
I'm trying to avoid implementing a custom server by using experimental `basePath` and `rewrites` features. However, when using together with `getServerSideProps`, the page reloads and throws a `TypoError` in the terminal.

To reproduce, add the following config to the [`with-redux` example](https://github.com/vercel/next.js/tree/canary/examples/with-redux).

```
// next.config.js
module.exports = {
  experimental: {
    basePath: "/t",
    async rewrites() {
      return [
        { source: "/abc", destination: "/ssr" },
      ];
    },
  },
}
```

and change `<Link href="/ssr">` in `nav.js` to `<Link href="/ssr" as="/abc">`.

Click on the SSR link. Observer the reload and the following error in the terminal

 ```
TypeError [ERR_INVALID_ARG_TYPE]: The "string" argument must be of type string or an instance of Buffer or ArrayBuffer. Received undefined
```

Any help is appreciated. Thanks!

**Comment by DumbledoreD** [comment]
I tried using `assetPrefix` in the config instead of `basePath`. I also put the next app behind Nginx where the "/t" base path is stripped when the request reaches the next server. However, the browser still does a hard refresh between pages.

**Comment by ijjk** [comment, marked as answer]
Hi, this should be resolved on the latest canary of Next.js `v9.4.5-canary.25` specifically [this PR](https://github.com/vercel/next.js/pull/14667). Note: on the latest canary these features are no longer nested under the `experimental` config

**Reply by DumbledoreD** [reply]
Sorry about the late reply. Can confirm it's fixed. Thanks!

**Reply by DumbledoreD** [reply]
A new problem arises and I'm on `v9.4.5-canary.33`. My `getServerSideProps` relies on the `context.req.url` to get the correct props from my backend. However, now it sees `/_next/data/development/pageName.json` instead of the URL in the `Link`'s `as` attribute. Is there a way to get the changing-to-location URL in `getServerSideProps`?

**Reply by ijjk** [reply]
The data request is currently expected to be to `/_next/data/development/pageName.json` so that it can be mapped to the correct page in all cases. If it were the rewritten `as` value it would fail to map to the page currently since the user would also need to know to add a rewrite for the `/_next/data` endpoint which adds complexity.

If you need to know the `as` value in `getServerSideProps` you can add a query value to the `href` e.g. `href="/page?as=/rewritten-path"`

**Reply by DumbledoreD** [reply]
That works. Thanks a lot!
